solutions/src/main/java/ru/sberbank/school/task02/Client.java

```diff
@@ -0,0 +1,140 @@
+package ru.sberbank.school.task02;
+
+import ru.sberbank.school.task02.exception.ConverterConfigurationException;
+import ru.sberbank.school.task02.exception.FxConversionException;
+import ru.sberbank.school.task02.exception.WrongSymbolException;
+import ru.sberbank.school.task02.util.*;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static ru.sberbank.school.task02.FxRequestConverter.*;
+
+/**
+ * Created by Mart
+ * 10.05.2019
+ **/
+public class Client implements FxClientController {
+    private FxConversionService conversionService;
+    private Beneficiary beneficiary;
+
+    public Client(FxConversionService conversionService, Beneficiary beneficiary) {
+        this.conversionService = conversionService;
+        this.beneficiary = beneficiary;
+    }
+
+    public Client( FxConversionService conversionService) {
+        this(conversionService, getBenificiary());
+    }
+
+    public static void main(String[] args) {
+        ServiceFactory serviceFactory = new ServiceFactoryImpl();
+        ExternalQuotesService externalQuotesService = new ExternalQuotesServiceDemo();
+        FxConversionService fxConversionService = serviceFactory.getFxConversionService(externalQuotesService);
+
+        Client client = new Client(fxConversionService);
+        FxRequest request = getFxRequestFromArgs(args);
+        FxResponse response = client.fetchResult(request);
+        System.out.println(request);
+        System.out.println(client.beneficiary);
+        System.out.println(response);
+    }
+
+    @Override
+    public List<FxResponse> fetchResult(List<FxRequest> requests) {
+        Objects.requireNonNull(requests);
+        List<FxResponse> resultList = new ArrayList<>();
+        for (FxRequest req : requests) {
+            resultList.add(fetchResult(req));
+        }
+        return resultList;
+    }
+
+    @Override
+    public FxResponse fetchResult(FxRequest requests) {
+        Objects.requireNonNull(requests);
+        BigDecimal amount = getAmount(requests);
+
+            Optional<BigDecimal> findPrice = Optional.of(conversionService.convert(getClientOperation(requests),
+                    getSymbol(requests), amount));
+
+            String dateTime = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
+
+            return findPrice.map(bigDecimal -> new FxResponse(requests.getSymbol(),
```

Comment:
&#39;method def&#39; child has incorrect indentation level 12, expected level should be 8.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1069612058>*